wp-content/themes/core/assets/js/admin/block-animation.js

```diff
@@ -0,0 +1,354 @@
+		} = attributes;
+
+		let blockClass =
+			attributes.className !== undefined ? attributes.className : '';
+		const blockStyles = { ...props.style };
+
+		if ( animationStyle !== undefined && animationStyle !== 'none' ) {
+			blockClass =
+				animationPosition !== undefined && animationPosition
+					? `${ blockClass } tribe-animation-style-${ animationStyle } is-animated-on-scroll-full`
+					: `${ blockClass } tribe-animation-style-${ animationStyle } is-animated-on-scroll`;
+
+			if ( animationSpeed !== undefined && animationSpeed ) {
+				blockStyles[ '--tribe-animation-speed' ] = animationSpeed;
+			}
+
+			if ( animationDelay !== undefined && animationDelay ) {
+				blockStyles[ '--tribe-animation-delay' ] = animationDelay;
+			}
+
+			if ( animationEasing !== undefined && animationEasing ) {
+				blockStyles[ '--tribe-animation-easing' ] = animationEasing;
+			}
+
+			if ( animationTrigger !== undefined && animationTrigger ) {
+				blockClass = `${ blockClass } tribe-animate-multiple`;
+			}
+		}
+
+		const blockProps = {
+			...props,
+			attributes: {
+				...attributes,
+				className: blockClass,
+			},
+			style: blockStyles,
+		};
+
+		return (
+			<Fragment>
+				<div style={ blockStyles } className={ blockClass }>
+					<BlockEdit { ...blockProps } />
+					{ isSelected && (
+						<InspectorControls>
+							<PanelBody
+								title="Tribe Animations"
+								initialOpen={ false }
+							>
+								<SelectControl
+									label="Animation Style"
+									value={ animationStyle ?? 'none' }
+									help="Animation style is the type of animation you want to display."
+									onChange={ ( newValue ) => {
+										setAttributes( {
+											animationStyle: newValue,
+										} );
+									} }
+									options={ [
+										{
+											label: __( 'None', 'tribe' ),
+											value: 'none',
+										},
+										{
+											label: __( 'Fade In', 'tribe' ),
+											value: 'fade-in',
+										},
+										{
+											label: __( 'Fade In Up', 'tribe' ),
+											value: 'fade-in-up',
+										},
+										{
+											label: __(
+												'Fade In Down',
+												'tribe'
+											),
+											value: 'fade-in-down',
+										},
+										{
+											label: __(
+												'Fade In Right',
+												'tribe'
+											),
+											value: 'fade-in-right',
+										},
+										{
+											label: __(
+												'Fade In Left',
+												'tribe'
+											),
+											value: 'fade-in-left',
+										},
+									] }
+								/>
+								<SelectControl
+									label="Animation Speed"
+									value={ animationSpeed ?? '0.3s' }
+									help="Animation speed is the speed at which the animation should run."
+									onChange={ ( newValue ) =>
+										setAttributes( {
+											animationSpeed: newValue,
+										} )
+									}
+									options={ [
+										{
+											label: __( 'Extra Slow', 'tribe' ),
+											value: '0.7s',
```

Comment:
Having the animation values hard-coded in here feels odd to me. Could we give control over to the admin user to define an array of animation speeds? The same applies to delays, and easing as well.

Alternatively, if we want to keep this under the hood, maybe we could start a constants file to contain all of this kind of knowledge in one location. That might make it easier to maintain down the road.